src/NUnitFramework/framework/Internal/TestProgressReporter.cs

```diff
@@ -34,27 +40,54 @@ public TestProgressReporter(ICallbackEventHandler handler)
         /// <param name="test">The test that is starting</param>
         public void TestStarted(ITest test)
         {
-            var parent = GetParent(test);
+            var message = CreateTestStartedMessage(test);
+
             try
             {
-                string report;
-                if (test is TestSuite)
+                _handler.RaiseCallbackEvent(message);
+            }
+            catch (Exception ex)
+            {
+                log.Error($"Exception processing {test.FullName}{Environment.NewLine}{ex}");
+            }
+        }
+
+        private string CreateTestStartedMessage(ITest test)
+        {
+            var parent = GetParent(test);
+
+            var stringBuilder = GetStringBuilder();
+
+            using var stringWriter = new StringWriter(stringBuilder);
+            using (var writer = XmlWriter.Create(stringWriter, XmlExtensions.FragmentWriterSettings))
+            {
+                var isSuite = test is TestSuite;
+                writer.WriteStartElement(isSuite ? "start-suite" : "start-test");
+
+                writer.WriteAttributeString("id", test.Id);
+                writer.WriteAttributeString("parentId", parent != null ? parent.Id : string.Empty);
+                writer.WriteAttributeStringSafe("name", test.Name);
+                writer.WriteAttributeStringSafe("fullname", test.FullName);
+                writer.WriteAttributeStringSafe("type", test.TestType);
+
+                if (isSuite)
```

Comment:
Thank you 😄 
I hadn't expected my earlier comment about this to make it into the PR but it's a welcome addition